packages/editor/src/client/blocks/mapLeafletBlock.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {useState} from 'react'
+import {Map, TileLayer, Marker, Popup} from 'react-leaflet'
+import 'leaflet/dist/leaflet.css'
+import '../customCSS/leaflet.css'
+
+import {Icon} from 'leaflet'
+
+// stupid hack so that leaflet's images work after going through webpack
+//@ts-ignore
+import marker from 'leaflet/dist/images/marker-icon.png'
+//@ts-ignore
+import marker2x from 'leaflet/dist/images/marker-icon-2x.png'
+//@ts-ignore
+import markerShadow from 'leaflet/dist/images/marker-shadow.png'
+
+//@ts-ignore
+delete Icon.Default.prototype._getIconUrl
+
+Icon.Default.mergeOptions({
+  iconRetinaUrl: marker2x,
+  iconUrl: marker,
+  shadowUrl: markerShadow
+})
+
+import {
+  BlockProps,
+  FieldProps,
+  Box,
+  Card,
+  PlaceholderInput,
+  ZIndex,
+  IconButton,
+  Spacing,
+  TypographicTextArea,
+  Drawer,
+  Image
+} from '@karma.run/ui'
+
+import {
+  MaterialIconImageOutlined,
+  MaterialIconEditOutlined,
+  MaterialIconClose
+} from '@karma.run/icons'
+
+import {ImageSelectPanel} from '../panel/imageSelectPanel'
+import {ImagedEditPanel} from '../panel/imageEditPanel'
+
+import {MapLeafletBlockValue, MapLeafletItem} from './types'
+
+// import {isFunctionalUpdate} from '@karma.run/react'
+
+export interface MapLeafletBlockProps extends BlockProps<MapLeafletBlockValue> {}
+
+export function MapLeafletBlock({value, onChange, disabled}: BlockProps<MapLeafletBlockValue>) {
+  const {zoom, centerLat, centerLng, caption} = value
+
+  return (
+    <>
+      <Map
+        center={[centerLat, centerLng]}
+        zoom={zoom}
+        style={{width: '100%', height: '45vh'}}
+        onViewportChange={e => {
+          if (e.center && e.zoom) {
+            onChange({...value, centerLat: e.center[0], centerLng: e.center[1], zoom: e.zoom})
+          }
+        }}>
+        <TileLayer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        />
+        <Marker position={[centerLat, centerLng]}></Marker>
+      </Map>
+      <Box marginTop={Spacing.ExtraSmall}>
+        <TypographicTextArea
```

Comment:
Why not use `<TextInput>`?

Reply:
I was taking inspiration from imageBlock's caption which uses ``<TypographicTextArea>`` but I certainly could also use ``<TextIinput>``. I'm not sure anymore what's the command to load the karma.run storyboard to see how to implement it; can you help me out?

Reply:
Clone the repository. `yarn install`, `yarn build` and `yarn watch:ui` and then under http://localhost:3010. It's all in the Readme of the repository.